logstash-core/spec/logstash/instrument/periodic_poller/dlq_spec.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/instrument/periodic_poller/dlq"
+require "logstash/instrument/collector"
+require "logstash/environment"
+require "logstash/agent"
+
+describe LogStash::Instrument::PeriodicPoller::DeadLetterQueue do
+  subject { LogStash::Instrument::PeriodicPoller::DeadLetterQueue }
+
+  let(:metric) { LogStash::Instrument::Metric.new(LogStash::Instrument::Collector.new) }
+  let(:agent) { double("agent")}
+  let(:pipeline) { double("pipeline")}
+  let(:pipelines) { { :main => pipeline } }
+  let(:options) { {} }
+  subject(:dlq) { described_class.new(metric, agent, options) }
+
+  it "should initialize cleanly" do
+    expect { dlq }.not_to raise_error
+  end
+
+
+
+  describe 'collect' do
+    subject(:collect) { dlq.collect }
+
+    before :each do
+      allow(agent).to receive(:with_running_pipelines) { |&block| block.call(pipelines) }
+      allow(pipeline).to receive(:collect_dlq_stats)
+    end
+
+    it 'should run cleanly' do
+      expect { collect }.not_to raise_error
+    end
+
+    it 'should request dlq stats from the pipeline' do
```

Comment:
This test asserts some deep assumptions about the names of methods and the way things are invoked.

Can this test be rewritten to test output (maybe the return value of `dlq.collect` instead of side effects such as what methods are called, etc?